File: banda/JCZhang-Banda/JCZhang-Banda-JAVA/src/main/java/factories/SingerFactory.java
package factories;


import models.Singer;

public class SingerFactory {
    public static int counter = 0;

    public static Singer createRandomSinger() {
        String[] names = {
                "Paquita",
                "Ernesto",
                "Antonio",
                "Alfonso",
                "Patricio",
                "BobEsponja",
                "Arenita",
                "DonCangrejo",
                "Calamardo",
                "Juan",
                "Laura",
                "Aitana",
                "Aitor",
                "Juanjo",
                "Paco"
        };

        int[] expYears = {
                2,
                4,
                25,
                6,
                8,
                54,
                56,
                3,
                6,
                27,
                4,
                1,
                7,
                36,
                0,
                13
        };

        String[] toneTypes = {
                "High",
                "Low"
        };

        String name = names[(int) (Math.random() * names.length)];
        int yearsOfExperience = expYears[(int) (Math.random() * expYears.length)];
        String tone = toneTypes[(int) (Math.random() * toneTypes.length)];
        counter++;
        return new Singer(name, yearsOfExperience, tone);
    }
}

File: banda/JCZhang-Banda/JCZhang-Banda_KOTLIN/src/main/kotlin/factories/SingerFactory.kt
package factories

import models.Singer

object SingerFactory {
    var singerCounter = 0
    fun createRandomSinger(): Singer {
        singerCounter++
        val names = arrayOf(
            "Paquita",
            "Ernesto",
            "Antonio",
            "Alfonso",
            "Patricio",
            "BobEsponja",
            "Arenita",
            "DonCangrejo",
            "Calamardo",
            "Juan",
            "Laura",
            "Aitana",
            "Aitor",
            "Juanjo",
            "Paco"
        )
        val expYears = intArrayOf(
            2,
            4,
            25,
            6,
            8,
            54,
            56,
            3,
            6,
            27,
            4,
            1,
            7,
            36,
            0,
            13
        )
        val toneTypes = arrayOf(
            "High",
            "Low"
        )
        val name = names[names.indices.random()]
        val yearsOfExperience = expYears[expYears.indices.random()]
        val tone = toneTypes[toneTypes.indices.random()]

        return Singer(name, yearsOfExperience, tone)
    }
}